### PR TITLE
Token and role validation

### PR DESCRIPTION
**Pull request by Lyhzinha**
User model changed: add role field;
Token creation when logging in;
Token and role validation when trying to access all users (only admins can do it);
Routes updated
Transforming "utils.js" into middleware.

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
     development: {
-        port: process.env.PORT || 3000,
+        port: process.env.PORT || 3005,
```
Comment (nunovinhas) [inline config.js:3]: Why this change?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,6 @@
   "description": "Some beautiful app",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (nunovinhas) [inline package.json:7]: Why this change?

File: config.js (hunk 1 above)
Comment (Lyhzinha) [reply config.js:3]: I have buildtoo working on 3000 port. That why I changed the port of linu

File: package.json (hunk 2 above)
Comment (Lyhzinha) [reply package.json:7]: This line was added by default and now I know that I won use it.
